public/app/core/utils/UserProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+
+export interface UserAPI {
+  changePassword: (ChangePassword: ChangePasswordFields) => void;
+}
+
+interface LoadingStates {
+  changePassword: boolean;
+}
+
+export interface ChangePasswordFields {
+  oldPassword: string;
+  newPassword: string;
+  confirmNew: string;
+}
+
+export interface Props {
+  children: (api: UserAPI, states: LoadingStates) => JSX.Element;
+}
+
+export interface State {
+  loadingStates: LoadingStates;
+}
+
+export class UserProvider extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
```

Comment:
to initiate state you do not need to define a constructor, you can just do like this:
https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/stackdriver/components/Help.tsx#L19